moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -538,6 +538,12 @@ bool Trajectory::integrateForward(std::list<TrajectoryStep>& trajectory, double
 
     if (next_discontinuity != switching_points.end() && path_pos > next_discontinuity->first)
     {
+      // Avoid having a TrajectoryStep with path_pos near a switching point which will cause an almost identical
+      // TrajectoryStep get added in the next run (https://github.com/ros-planning/moveit/issues/1665)
+      if (path_pos - next_discontinuity->first < time_step_ * 1e-5)
+      {
+        continue;
+      }
       path_vel = old_path_vel +
                  (next_discontinuity->first - old_path_pos) * (path_vel - old_path_vel) / (path_pos - old_path_pos);
       path_pos = next_discontinuity->first;
```

Comment:
As an alternative 
```suggestion
      path_pos = next_discontinuity->first;
      if (std::abs((path_vel - old_path_vel) / (path_pos - next_discontinuity->first)) > maximum_acceleration_)
          continue;      
```

Reply:
This results in immediately dividing by zero

Reply:
My bad. If you moved this before the assignment would it still not work?

Reply:
You would need to set `maximum_acceleration_` to something reasonable